### PR TITLE
[IMP][9.0] Partner names order configurable

### PR DESCRIPTION
**Pull request by lasley**
This is a forward port of `partner_firstname` changes from #227. Basically just a cherry-pick and a few style adjustments as noted below:

* Prefer odoo to openerp & remove data tags
* Move exceptions file out of models

### CODE REVIEW

File: partner_firstname/README.rst [hunk 1]
```diff
@@ -16,22 +37,40 @@ and the *first name*. This avoids breaking compatibility with other modules.
...
 For the same reason, after installing, previous names for contacts will stay in
 the *name* field, and the first time you edit any of them you will be asked to
 supply the *last name* and *first name* (just once per contact).
 
-You can use *_get_inverse_name* method to get lastname and firstname from a simple string
-and also *_get_computed_name* to get a name form the lastname and firstname. 
-These methods can be overridden to change the format specified above
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/134/8.0
```
Comment (pedrobaeza) [inline partner_firstname/README.rst]: s/8.0/9.0

File: partner_firstname/README.rst [hunk 2]
```diff
@@ -16,22 +37,40 @@ and the *first name*. This avoids breaking compatibility with other modules.
...
+
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues
+<https://github.com/OCA/partner-contact/issues>`_. In case of trouble, please
+check there if your issue has already been reported. If you spotted it first,
+help us smashing it by providing a detailed and welcomed feedback `here
+<https://github.com/OCA/partner-contact/issues/new?body=module:%20partner_firstname%0Aversion:%208.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (pedrobaeza) [inline partner_firstname/README.rst]: Remove this part

**Review by pedrobaeza — APPROVED** [review]
Minor changes, but overall :+1: 

Well, when Travis goes green

File: partner_firstname/views/base_config_view.xml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+<!-- © 2015 Antiun Ingenieria S.L. - Antonio Espinosa
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html). -->
+<odoo>
+
+<record id="view_general_configuration" model="ir.ui.view">
+    <field name="name">Add partner_names_order config parameter</field>
+    <field name="model">base.config.settings</field>
+    <field name="inherit_id" ref="base_setup.view_general_configuration"/>
+    <field name="arch" type="xml">
+        <xpath expr="//label[@string='Email']/.." position='after'>
```
Comment (lasley) [inline partner_firstname/views/base_config_view.xml]: No string selector in >8

Comment (pedrobaeza) [reply partner_firstname/views/base_config_view.xml]: Yeah, that's it. Let's hope there's a new shining name attribute around...

**Comment by lasley** [comment]
Pesky rebase got me on this one - missed an import! Alright we're good now & Pedro's comments attended to, I'll squash up before we merge. 

I'll be submitting the v10 in a bit as well, that's the version I really need it on 😉

File: partner_firstname/views/base_config_view.xml (hunk 3 above)
Comment (lasley) [reply partner_firstname/views/base_config_view.xml]: There was! `email_label` 😄

**Comment by pedrobaeza** [comment]
Thanks for taking care of this version.

**Comment by lasley** [comment]
Added icon fix from https://github.com/OCA/partner-contact/pull/334#issuecomment-275819078

**Comment by whnr** [comment]
Do I understand correctly that this is not available in the App? How can I use it then? Every search for this topic points here but it seems to be not available in the released version in Odoo.

**Comment by lasley** [comment]
@floweh This is not available in the v10 version until someone else adds an accepted review, which will allow for a merge

**Comment by whnr** [comment]
@lasley yes, but this is the V9 I need… The changes seem to be approved.

**Comment by lasley** [comment]
Sorry I meant v9, which is the version of this PR. 

> The changes seem to be approved.

A change is not approved without two reviews, [per the guidelines](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#review). It would be amazing if you went ahead and provided one, so that we could look at merging this.

**Comment by pedrobaeza** [comment]
As this is has been long outstanding and it has my bless, we can merge it and remove pending tasks.

**Pull request merged**